lib/travis_event/handlers/stalled_finished_job.rb

```diff
@@ -0,0 +1,116 @@
+require 'travis/client'
+
+module TravisEvent
+  module Handlers
+    class StalledFinishedJob
+      include Sidekiq::Worker
+      sidekiq_options :queue => :travis_event, :retry => 5
+
+      STALLED_BUILD_TEXT = "\n\nNo output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.\n\nThe build has been terminated\n\n"
+      HANDLED_EVENTS  = ['job:finished'].freeze
+      ERROR = "errored".freeze
+
+      attr_reader :repo, :repo_name, :number, :event_type, :state
+
+      def perform(repo_name, number, event_type, state, raw_branch_name, pull_request)
+        @repo_name    = repo_name
+        @number       = number
+        @event_type   = event_type
+        @state        = state
+        @pull_request = pull_request
+
+        return if skip_based_on_args?
+
+        # Local checks: skip unknown repos or non-current branches
+        return unless current_commit_monitor_branch(raw_branch_name)
+
+        # Remote checks: Skip missing travis repo or job and non-stalled builds
+        return unless travis_repo = Travis::Repository.find(repo_name)
+        return unless @job = find_job(travis_repo, number)
+        return unless stalled?
+
+        message = "Restarting stalled job: [#{repo_name}##{number}]"
+        commit_monitor_branch.write_github_comment(message)
```

Comment:
@Fryguy notice, I added a placeholder for commenting on a PR CommitMonitorBranch. It seemed like the place to do it.